libraries/AP_HAL_ChibiOS/Util.cpp

```diff
@@ -372,22 +372,23 @@ Util::FlashBootloader Util::flash_bootloader()
 /*
   display system identifer - board type and serial number
  */
-bool Util::get_system_id(char buf[40])
+bool Util::get_system_id(char buf[50])
 {
     uint8_t serialid[12];
-    char board_name[14];
+    char board_name[24];
 
     memcpy(serialid, (const void *)UDID_START, 12);
-    strncpy(board_name, CHIBIOS_SHORT_BOARD_NAME, 13);
-    board_name[13] = 0;
+    static_assert(sizeof(CHIBIOS_SHORT_BOARD_NAME) <= 24, "CHIBIOS_SHORT_BOARD_NAME must be 23 characters or less");
```

Comment:
```suggestion
    static_assert(sizeof(CHIBIOS_SHORT_BOARD_NAME) < 24, "CHIBIOS_SHORT_BOARD_NAME must be 23 characters or less");
```

Reply:
Recommend less than/equal to here to account for null character string terminator (I tested this).

Reply:
Ah, `strlen` vs `sizeof`. I don't think strlen would work in a assert, but it might. Might also be worth a comment.